### ISSUE TITLE
When to stop training and how to determine best model?

### ISSUE THREAD

**Issue by namhhn** [mazurowski-lab/segmentation-guided-diffusion#20]
It is stated in the paper that you don't use FID to evaluate the generated images so I am curious about the criteria you chose the best model for generating new data for the segmentation task. I am assuming we test the segmentation model with generated data from the model at certain epoch and the generated set that yields highest dice score will be from the best model?
Thanks for sharing the repo!

**Comment by nickk124** [comment]
Hi, thanks to diffusion models typically having stable training convergence (unlike GANs and their more messy training objective), we simply trained each diffusion model in the paper until convergence/the loss stopped noticeably decreasing; 400 epochs for all models. 

For the segmentation models used, we selected the final checkpoint based off of best validation set performance/Dice (the validation set of held-out real or synthetic images, depending on the model)
